include/util/json_deep_compare.hpp

```diff
@@ -0,0 +1,158 @@
+#ifndef UTIL_JSON_DEEP_COMPARE_HPP
+#define UTIL_JSON_DEEP_COMPARE_HPP
+
+#include "util/json_container.hpp"
+#include "util/integer_range.hpp"
+
+#include <boost/assert.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <set>
+
+namespace osrm
+{
+namespace util
+{
+namespace json
+{
+
+struct Comparator : mapbox::util::static_visitor<bool>
```

Comment:
Can't we just provide `operator==` for out types in json_container and then use mapbox/variant's `operator==`? https://github.com/mapbox/variant/blob/3f025adbf599d8dd9bfca02d45b37e49a2cae841/variant.hpp#L813-L829